tests/e2e/contracts/probes/observability_probe.py

```
@@ -638,30 +637,47 @@ async def probe_span_attributes(self, trace_id: str, attributes: dict[str, Any])
                 for item in latest.get("hits", []):
                     if item.get("trace_id") == generated_trace_id:
-                        payload = item
+                        latest_payload = item
                         break
-                if payload:
+                if latest_payload:
+                    trace_payload = await self._get_trace(generated_trace_id)
+                    span_payloads = trace_payload.get("spans", [])
                     break
                 await asyncio.sleep(1.0)
         except Exception as exc:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)